app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java

```diff
@@ -79,154 +102,169 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
+//                .title("London"));
+//        LatLng latLng = new LatLng(51.5072, 0.1276);
+//        map.moveCamera(CameraUpdateFactory.newLatLng(latLng));
+//        map.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng,15f));
+    }
+
+    private void getLocation() {
+        progressDialog.show();
+        if(Build.VERSION.SDK_INT>=23){
+            if(checkPermission()){
+                getDeviceLocation();
+            }
+            else{
+                requestPermission();
+            }
+        }
+        else{
+            getDeviceLocation();
+        }
+    }
+
+    private void requestPermission() {
+        ActivityCompat.requestPermissions(getActivity(),permissions_all,PERMISSION_CODE);
+    }
+
+    private boolean checkPermission() {
+        for(int i=0;i<permissions_all.length;i++){
+            int result= ContextCompat.checkSelfPermission(getContext(),permissions_all[i]);
+            if(result== PackageManager.PERMISSION_GRANTED){
+                continue;
+            }
+            else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private void getDeviceLocation() {
+        locationManager=(LocationManager) getActivity().getSystemService(Service.LOCATION_SERVICE);
+        isGpsLocation=locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+        isNetworklocation=locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+        if(!isGpsLocation && !isNetworklocation){
+            showSettingForLocation();
+            getLastlocation();
+        }
+        else{
+            getFinalLocation();
+        }
+    }
+
+    private void getLastlocation() {
+        if(locationManager!=null) {
+            try {
+                Criteria criteria = new Criteria();
+                String provider = locationManager.getBestProvider(criteria,false);
+                Location location=locationManager.getLastKnownLocation(provider);
+            } catch (SecurityException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        switch (requestCode){
+            case PERMISSION_CODE:
+                if(grantResults.length>0 && grantResults[0]== PackageManager.PERMISSION_GRANTED){
+                    getFinalLocation();
+                }
+                else{
+                    Toast.makeText(getContext(), "Permission Failed", Toast.LENGTH_SHORT).show();
+                }
+        }
+    }
+
+    private void getFinalLocation() {
+        //one thing i missed in permission let's complete it
+        try{
+            if(isGpsLocation){
+                locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER,1000*60*1,10, (LocationListener) getContext());
```

Comment:
Here's another example of consting:
MILLIS_IN_ONE_SECOND = 1000;
SECONDS_IN_ONE_MINUTE = 60;
MILLIS_IN_ONE_MINUTE = SECONDS_IN_ONE_MINUTE * MILLIS_IN_ONE_SECOND;
????? = 10;

Although, I'm pretty sure these consts are already defined somewhere in java, and you dont need to redefine them.